packages/app-builder-lib/src/node-module-collector/index.ts

```diff
@@ -3,14 +3,34 @@ import { PnpmNodeModulesCollector } from "./pnpmNodeModulesCollector"
 import { YarnNodeModulesCollector } from "./yarnNodeModulesCollector"
 import { detect, PM, getPackageManagerVersion } from "./packageManager"
 import { NodeModuleInfo } from "./types"
+import { exec } from "builder-util"
+
+async function isPnpmProjectHoisted(rootDir: string) {
+  const command = process.platform === "win32" ? "pnpm.cmd" : "pnpm"
+  const config = await exec(command, ["config", "list"], { cwd: rootDir, shell: true })
+  const lines = config
```

Comment:
I believe you can just do this

```suggestion
  const lines = Object.fromEntries(
  config.split("\n").map(line =>
    line.split("=").map(s => s.trim())
  )
);

```

This keeps the code more maintainable I think

Reply:
That's beautiful. Will update the PR